src/components/button/button.js

```diff
@@ -33,12 +34,52 @@ Button.prototype.handleKeyDown = function (event) {
   }
 }
 
+/**
+* If the click quickly succeeds a previous click then nothing will happen.
+* This stops people accidentally causing multiple form submissions by
+* double clicking buttons.
+*/
+Button.prototype.debounce = function (event) {
+  var target = event.target
+  // Check the button that is clicked on has the preventDoubleClick feature enabled
+  if (target.attributes && target.getAttribute('data-prevent-double-click') !== 'true') {
+    return
+  }
+
+  // Only handle events on <input> and <button> elements
+  var tagName = event.target.tagName
+  if (tagName !== 'INPUT' && tagName !== 'BUTTON') {
+    return
+  }
+
+  var $button = event.target
+
+  // We only want to to handle submit buttons that are used in forms
```

Comment:
If this is now an opt-in behaviour, do we need to be as strict here?

If users add `data-prevent-double-click`, I think they'd expect it to work, regardless of whether it's in a form / a submit button?

Reply:
Yeah could probably simplify this now...